public/src/router/Routes.tsx

```diff
@@ -1,13 +1,27 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import { Route, Switch, Redirect } from 'react-router-dom';
 import { RoutePath } from './RoutePath';
 import Main from '../pages/main/Main';
 import Login from '../pages/auth/Login/Login';
 import Register from '../pages/auth/Register/Register';
 import Marketplace from '../pages/marketplace/Marketplace';
+import Userprofile from 'src/pages/main/Userprofile';
+import AdminPage from 'src/pages/auth/AdminPage';
+import { getUserData } from 'src/api/httpClient';
 
 export const Routes: FC = () => {
   const user = sessionStorage.getItem('email');
+  const [isAdminUser, setIsAdminUser] = useState<boolean>(true);
+  useEffect(() => {
+    checkIsAdmin();
+  }, [isAdminUser]);
+
+  const checkIsAdmin = () => {
+    user
+      ? getUserData(user).then((data) => setIsAdminUser(data.isAdmin))
+      : null;
+  };
```

Comment:
The function name does not reflect that it actually does. Please rename it using the more clear name